galasa-managers-parent/galasa-managers-common-parent/dev.galasa.common/src/test/java/dev/galasa/SSLTLSContextNameTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.common;
+
+import org.junit.Test;
+
+import static org.junit.Assert.* ;
+import java.util.Properties;
+
+public class SSLTLSContextNameTest {
+    @Test
+    public void testNonIBMJVMGivesTLS12Context() throws Exception {
+        Properties testProps = new Properties();
+        testProps.setProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY,"FakeJavaRuntimeVendorName" );
+        // When...
+        String contextName = SSLTLSContextName.getSelectedSSLContextName(testProps);
+        // Then...
+        String expected = "TLSv1.2";
+        assertEquals(expected,contextName);
+    }
+    @Test
+    public void testIBMJVM8GivesSSL_TLSv2Context() throws Exception {
+        // Given...
+        Properties testProps = new Properties();
+        testProps.setProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY,"IBMJavaRuntimeVendorName" );
+        testProps.setProperty(SSLTLSContextName.JAVA_VERSION_PROPERTY, "8.0.16");
+        // When...
+        String contextName = SSLTLSContextName.getSelectedSSLContextName(testProps);
+        // Then...
+        String expected = "SSL_TLSv2";
+        assertEquals(expected,contextName);
+    }
+    
+    @Test
+    public void testIBMJVM11GivesTLSv12Context() throws Exception {
+        // Given...
+        Properties testProps = new Properties();
+        testProps.setProperty(SSLTLSContextName.JAVA_VENDOR_PROPERTY,"IBMJavaRuntimeVendorName" );
+        testProps.setProperty(SSLTLSContextName.JAVA_VERSION_PROPERTY, "11.0.16.1+1");
+        // When...
+        String contextName = SSLTLSContextName.getSelectedSSLContextName(testProps);
+        // Then...
+        String expected = "TLSv1.2";
+        assertEquals(expected,contextName);
+    }
+
+    @Test
```

Comment:
This test is kinda missing the point of unit testing.
The idea is to mimick things with mocks, or test inputs.
Not actually replicate the logic and make sure the code under test does the same thing the implementation code does. Happy to discuss...